src/actions/mintNFT.ts

```diff
@@ -1,3 +1,152 @@
-export const mintNFT = async () => {
-  throw new Error("It's not implemented");
+import { MintLayout, TOKEN_PROGRAM_ID, Token } from '@solana/spl-token';
+import { Keypair, PublicKey } from '@solana/web3.js';
+import { Connection } from './../Connection';
+import { MintTo, CreateAssociatedTokenAccount, CreateMint, Transaction } from './../programs';
+import {
+  CreateMasterEdition,
+  CreateMetadata,
+  Creator,
+  MasterEdition,
+  Metadata,
+  MetadataDataData,
+} from './../programs/metadata';
+import { Wallet } from 'src/wallet';
+import { ASSOCIATED_TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import { sendTransaction } from './transactions';
+import { lookup } from './../utils/metadata';
+import BN from 'bn.js';
+
+interface MintNFTParams {
+  connection: Connection;
+  wallet: Wallet;
+  uri: string;
+  maxSupply: number;
+}
+
+interface MintNFTResponse {
+  txId: string;
+  mint: PublicKey;
+  metadata: PublicKey;
+  edition: PublicKey;
+}
+
+export const mintNFT = async ({
+  connection,
+  wallet,
+  uri,
+  maxSupply,
+}: MintNFTParams): Promise<MintNFTResponse> => {
```

Comment:
Thanks, @m-sebastiyan for the quick review. Another question I didn't get in the original description was about supporting batch minting. Instead of accepting one metadata url minNFT would take a list. The instructions are bundled into a single transaction.

```javascript
mintNFT({
  connection,
   wallet,
   nfts: [
     { maxSupply: 0, uri: "http://arweave.net/dec" },
     { maxSupply: 10, uri: "https://arweave.net/xyz"}
   ]
})
```
First, what are your (and other) thoughts on the idea? And if you like are there any limitations I should be aware of. Like max number of instructions to send in a single transaction.

Reply:
I agree with that, would be pretty convenient! One thing is the transaction size, which you've mentioned, this needs to be considered. Nothing else comes to my mind.

Reply:
Even bundling 2 nft instruction sets into a single transaction exceeds the size limit so going to go forward with the single mint.
```
{                                                                                                                                                                                                  
      txs: [
        Transaction {
          signatures: [],
          feePayer: undefined,
          instructions: [Array],
          recentBlockhash: undefined,
          nonceInfo: undefined
        },
        Transaction {
          signatures: [],
          feePayer: undefined,
          instructions: [Array],
          recentBlockhash: undefined,
          nonceInfo: undefined
        }
      ],
      signers: [ Keypair { _keypair: [Object] }, Keypair { _keypair: [Object] } ]
    }

      at src/actions/mintNFT.ts:217:11

 FAIL  test/actions/mintNFT.test.ts
  minting an NFT
    when can find metadata json
      ✕ generates a unique mint and creates metadata plus master edition from metadata URL and max supply (2221 ms)                                                                                               

  ● minting an NFT › when can find metadata json › generates a unique mint and creates metadata plus master edition from metadata URL and max supply                                                              

    Transaction too large: 1356 > 1232

```